test/control_filters/test_gravity_compensation.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2023, Stogl Robotics Consulting UG (haftungsbeschränkt)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "test_gravity_compensation.hpp"
+#include <vector>
+
+TEST_F(GravityCompensationTest, TestGravityCompensationMissingParameters)
+{
+  std::shared_ptr<filters::FilterBase<geometry_msgs::msg::WrenchStamped>> filter_ =
+    std::make_shared<control_filters::GravityCompensation<geometry_msgs::msg::WrenchStamped>>();
+
+  node_->declare_parameter("world_frame", "world");
+
+  // one mandatory param missing, should fail
+  ASSERT_FALSE(filter_->configure("", "TestGravityCompensation",
+    node_->get_node_logging_interface(), node_->get_node_parameters_interface()));
+  /* NOTE: one cannot declare or set the missing param afterwards, to then test if configure works,
+   * because the param is read only and cannot be set anymore.
+   */
+}
+
+TEST_F(GravityCompensationTest, TestGravityCompensationParameters)
+{
+  std::shared_ptr<filters::FilterBase<geometry_msgs::msg::WrenchStamped>> filter_ =
+    std::make_shared<control_filters::GravityCompensation<geometry_msgs::msg::WrenchStamped>>();
+
+  double gravity_acc = 9.81;
+  double mass = 5.0;
+  node_->declare_parameter("world_frame", "world");
+  node_->declare_parameter("sensor_frame", "sensor");
+  node_->declare_parameter("CoG.force", std::vector<double>({0.0, 0.0, -gravity_acc * mass}));
```

Comment:
We shouldn't do this. We have generate_parameter_library to take care of this things. So no declaration of setting of parameters, but using generate_parameter_library that sets the params when node is started.

Reply:
For differnet cases simply use different `*.yaml` and test files

Reply:
hum, ok. I did not know this change to `generate_parameter_library` had to also be applied to the tests which are quite flexible with node parameter declaration. 
Indeed, in other PRs where I had to create tests with a YAML, the values in the YAML have to match the values in the test (to verify the values were loaded correctly by code). This means 2 files have to be edited when setting the parameters, to ensure everything matches. It is easier when params and verified in one place. 
 Maybe I could move to "set_parameter" only instead of declaring. But I think this might be complex or impossible because param declaration happens in the filter_chain, with the node passed to it, and should already have the params overriden there. To override the params in the node, they must be declared isn't it ? this is a chicken-egg problem.

Reply:
I did the change. I am not sure this is was really needed, but at least there is no "visible" declaration.